tests/test_smoke.py

```diff
@@ -1505,13 +1505,17 @@ def test_multi_hostname(generic_cloud: str):
     run_one_test(test)
 
 
-@pytest.mark.no_scp
+@pytest.mark.no_scp  # SCP does not support num_nodes > 1 yet
 def test_multi_node_failure(generic_cloud: str):
     name = _get_cluster_name()
     test = Test(
         'multi_node_failure',
         [
-            f'sky launch -y -c {name} --cloud {generic_cloud} tests/test_yamls/failed_worker_setup.yaml && exit 1',  # Ensure the job setup failed.
+            # TODO(zhwu): we use multi-thread to run the commands in setup
+            # commands in parallel, which makes it impossible to fail fast
+            # when one of the nodes fails. We should fix this in the future.
+            # Refer to the comment in `subprocess_utils.run_in_parallel`.
+            # f'sky launch -y -c {name} --cloud {generic_cloud} tests/test_yamls/failed_worker_setup.yaml && exit 1',  # Ensure the job setup failed.
```

Comment:
A bit confused here. It seems to say failed worker setup won't immediately fail the entire setup, right? But then tests/test_yamls/failed_worker_setup.yaml has a `sleep 10000` in setup?

Reply:
The issue only happens when `--detach-setup` is not provided. When setup is detached, we submit the setup to the remote machine as a job, which does not have the issue with multi-thread. I updated the comment. : )